### DISCUSSION TITLE
Public read-only indexes

### DISCUSSION THREAD

**Discussion by david-perez** [meilisearch/product#369] [Feedback & Feature Proposal]
I have read through the docs on [security and permissions](https://docs.meilisearch.com/reference/features/security.html#protecting-a-meilisearch-instance) and it seems like there's no middle-ground between disabling key-based authorization entirely and having API keys required to perform actions.

I'd like to make an index publically searchable without authorization, but disallow modification actions on it without an API key.

**Comment by gmourier** [comment]
Hi @david-perez 👋

Indeed, this is not possible today. Can you tell me a bit more about why do you need that? I'm assuming you don't have a unified interface and share the API to possible consumers?

**Reply by david-perez** [reply]
@gmourier That's right, Meilisearch's search dashboard suffices for my needs (aside: well, on mobile the UI is very broken, don't know if there's an issue for that) and exposing it to the public while disallowing indexes' modifications is all I want.

**Reply by gmourier** [reply]
Hello @david-perez!

Why don't you want / can't use the Authorization header?

You can create an API Key that can just search on some specific indexes thus not allowing any modification on it with the `/keys` endpoints since the v0.25 version! 

> (aside: well, on mobile the UI is very broken, don't know if there's an issue for that)

Thanks for the feedback, we have a dedicated repository for the mini-dashboard. https://github.com/meilisearch/mini-dashboard/issues. I could create an issue if you have more details to share about it. 🤓

**Reply by david-perez** [reply]
> You can create an API Key that can just search on some specific indexes thus not allowing any modification on it with the /keys endpoints since the v0.25 version!

That's great, but it still requires me to share an API key with users. What I'd like to do is have an index be publically searchable _by default_ (i.e. no API keys required to search it, think _anyone_ can access the dashboard), and at the same time disallow modification actions on the index.

> Why don't you want / can't use the Authorization header?

That's just another form of secret that I'd have to share with users upfront.

> Thanks for the feedback, we have a dedicated repository for the mini-dashboard. https://github.com/meilisearch/mini-dashboard/issues. I could create an issue if you have more details to share about it.

No specific feedback, it's in general very broken. I've cut https://github.com/meilisearch/mini-dashboard/issues/235.
